sdk/turing/session.py

```diff
@@ -55,12 +55,11 @@ def __init__(
         self._api_client = ApiClient(config)
 
         if use_google_oauth:
-            import google.auth
+            from caraml_auth.id_token_credentials import get_default_id_token_credentials
             from google.auth.transport.requests import Request
             from google.auth.transport.urllib3 import urllib3, AuthorizedHttp
 
-            # Load default credentials
-            credentials, _ = google.auth.default(scopes=TuringSession.OAUTH_SCOPES)
+            credentials = get_default_id_token_credentials(target_audience="turing-sdk.caraml")
```

Comment:
How is the `target_audience` expected to be used on the auth server? Or is it currently unused?

Reply:
It is unused right now. But we can set up the `RequestAuthentication` resource to validate the `aud` field which this `target_audience` field would translate into in the jwt token generated for service accounts. CMIIW @mbruner

Reply:
I see.. I wonder if we should simply call it "sdk.caraml" in order to not distinguish each SDK.

Reply:
I'm fine with any name to be honest 😅 `sdk.caraml` it shall be then!